test/Event/EventCalendarProjectorTest.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace CultuurNet\UDB3\Event;
+
+use \CultureFeed_Cdb_Data_Calendar_PeriodList as PeriodList;
+use \CultureFeed_Cdb_Data_Calendar_Period as Period;
+use \CultureFeed_Cdb_Data_Calendar_TimestampList as TimestampList;
+use \CultureFeed_Cdb_Data_Calendar_Timestamp as Timestamp;
+use CultuurNet\UDB3\Event\ReadModel\CalendarRepositoryInterface;
+use Doctrine\Common\Cache\ArrayCache;
+
+class EventCalendarProjectorTest extends CdbXMLProjectorTestBase
+{
+    /**
+     * @var CalendarRepositoryInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $repository;
+
+    /**
+     * @var EventCalendarProjector
+     */
+    protected $projector;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->repository = $this->getMock(CalendarRepositoryInterface::class);
+        $this->projector = new EventCalendarProjector($this->repository);
+    }
+
+    /**
+     * @test
+     */
+    public function it_saves_the_calendar_periods_from_events_imported_from_udb2()
+    {
+        $event = $this->eventImportedFromUDB2('samples/event_with_calendar_periods.cdbxml.xml');
+        $this->repositoryExpectsPeriodList();
+        $this->projector->applyEventImportedFromUDB2($event);
+    }
+
+    /**
+     * @test
+     */
+    public function it_saves_the_calendar_periods_from_events_updated_from_udb2()
+    {
+        $event = $this->eventUpdatedFromUDB2('samples/event_with_calendar_periods.cdbxml.xml');
+        $this->repositoryExpectsPeriodList();
+        $this->projector->applyEventUpdatedFromUDB2($event);
+    }
+
+    private function repositoryExpectsPeriodList()
+    {
+        $this->repository->expects($this->once())
+            ->method('save')
+            ->with(
+                'someId',
+                $this->callback(function (PeriodList $calendar) {
+                    // Counting the objects in an iterator will loop over all objects and set the pointer to a non-
```

Comment:
I find it better readable to instantiate a new PeriodList object here and add the properties you expect, then verify that it equals the one that was passed in. This also proofs that there did not happen any additional side effects on the PeriodList before it was saved to the repository.